src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  const stateCopy = { ...state };
+
+  for (const a of actions) {
+    switch (a.type) {
```

Comment:
Use a more desciptive name for the variable and use destructuring
```suggestion
  for (const action of actions) {
    const { type, ....... } = action;
    
    switch (type) {
```